src/main/java/duke/managers/Parser.java

```diff
@@ -0,0 +1,56 @@
+package duke.managers;
+
+import duke.commands.*;
+import duke.exceptions.DukeException;
+
+import java.util.ArrayList;
+
+/**
+ * Represents a manager that handles recognition of user input.
+ */
+public class Parser {
+
+    protected ArrayList<Command> commandList;
```

Comment:
Maybe Plural form should be used on names representing a collection of objects? e.g. commands